Lib/fontTools/ttLib/ttFont.py

```diff
@@ -499,48 +575,50 @@ def _readTable(self, tag):
             self._tableCache[(tag, data)] = table
         return table
 
-    def __setitem__(self, tag, table):
+    def __setitem__(self, tag: str, table: DefaultTable) -> None:
         self.tables[Tag(tag)] = table
 
-    def __delitem__(self, tag):
+    def __delitem__(self, tag: str) -> None:
+        tag = Tag(tag)
         if tag not in self:
             raise KeyError("'%s' table not found" % tag)
         if tag in self.tables:
             del self.tables[tag]
         if self.reader and tag in self.reader:
             del self.reader[tag]
 
-    def get(self, tag, default=None):
+    def get(self, tag: str, default: Any | None = None) -> DefaultTable | Any | None:
```

Comment:
it might be nice to `@overload` this, and use a `TypeVar` for the default, similar to `Mapping.get` in https://github.com/python/typeshed/blob/aa52024657e7aecfe03d08a7e3719a2ded9a907a/stdlib/typing.pyi#L768-L773